### PR TITLE
fix Solaris OS detection in config.pm

### PR DESCRIPTION
**Pull request by jenda1**
This fixes openssl/openssl#15435 issue.

This change also fixes the Sun compiler detection - Oracle Developer Studio is the current name for the compilers that were known as Sun Workshop long ago.

### CODE REVIEW

**Comment by jenda1** [comment]
I sent the ICLA form to o [legal](mailto:legal@opensslfoundation.org) just now.

**Comment by mattcaswell** [comment]
Close/reopen to kick clabot

File: util/perl/OpenSSL/config.pm [hunk 1]
```diff
@@ -238,7 +240,7 @@ sub get_sco_type {
 sub guess_system {
     ($SYSTEM, undef, $RELEASE, $VERSION, $MACHINE) = POSIX::uname();
     my $sys = "${SYSTEM}:${RELEASE}:${VERSION}:${MACHINE}";
-
+
```
Comment (t8m) [inline util/perl/OpenSSL/config.pm:243]: Please drop this spurious whitespace addition.

**Review by t8m — COMMENTED** [review]
Please do not use tabs to indent.

**Review by levitte — APPROVED** [review]

**Review by t-j-h — APPROVED** [review]

**Comment by levitte** [comment]
Agreeing on urgent

**Comment by levitte** [comment]
Merged

8d67621de16990132c13f6a11bcc18ce8e9cdd47 fix Solaris OS detection in config.pm

**Pull request closed without merging**